src/com/magento/idea/magento2plugin/actions/generation/generator/CLICommandClassGenerator.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+package com.magento.idea.magento2plugin.actions.generation.generator;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiDirectory;
+import com.intellij.psi.PsiFile;
+import com.jetbrains.php.lang.psi.PhpFile;
+import com.magento.idea.magento2plugin.actions.generation.data.CLICommandClassData;
+import com.magento.idea.magento2plugin.actions.generation.generator.util.DirectoryGenerator;
+import com.magento.idea.magento2plugin.actions.generation.generator.util.FileFromTemplateGenerator;
+import com.magento.idea.magento2plugin.bundles.ValidatorBundle;
+import com.magento.idea.magento2plugin.indexes.ModuleIndex;
+import com.magento.idea.magento2plugin.magento.files.CLICommandTemplate;
+import com.magento.idea.magento2plugin.bundles.CommonBundle;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Properties;
+
+public class CLICommandClassGenerator extends FileGenerator {
+    private final CLICommandClassData cliCommandClassData;
+    private final Project project;
+    private final ValidatorBundle validatorBundle;
+    private final DirectoryGenerator directoryGenerator;
+    private final FileFromTemplateGenerator fileFromTemplateGenerator;
+    private final CommonBundle commonBundle;
+
+    public CLICommandClassGenerator(Project project, @NotNull CLICommandClassData cliCommandClassData) {
+        super(project);
+        this.project = project;
+        this.cliCommandClassData = cliCommandClassData;
+
+        this.directoryGenerator = DirectoryGenerator.getInstance();
+        this.fileFromTemplateGenerator = FileFromTemplateGenerator.getInstance(project);
+        this.validatorBundle = new ValidatorBundle();
+        this.commonBundle = new CommonBundle();
+    }
+
+    public PsiFile generate(String actionName) {
+        PhpFile cliCommandFile = createCLICommandClass(actionName);
+
+        if (cliCommandFile == null) {
+            String errorMessage = validatorBundle.message(
+            "validator.file.cantBeCreated",
+            commonBundle.message("common.cli.class.title")
```

Comment:
Let's have some indents here
```suggestion
                "validator.file.cantBeCreated",
                commonBundle.message("common.cli.class.title")
```